crates/extensions/tedge_flows/src/actor.rs

```diff
@@ -156,28 +198,125 @@ impl FlowsMapper {
+                self.on_process_eos(&topic).await?
+            }
+        }
+        Ok(())
+    }
+
+    async fn on_process_message(
+        &mut self,
+        flow_name: String,
+        line: String,
+    ) -> Result<(), RuntimeError> {
+        if let Some(flow) = self.processor.flows.get(&flow_name) {
+            let topic = flow.input.enforced_topic().unwrap_or_default();
+            let source = SourceTag::Process {
+                flow: flow_name.clone(),
+            };
+            self.on_message(source, Message::new(topic, line)).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn on_process_eos(&mut self, flow_name: &str) -> Result<(), RuntimeError> {
+        if let Some(flow) = self.processor.flows.get(flow_name) {
+            if let Some(request) = flow.watch_request() {
+                info!(target: "flows", "Reconnecting input: {flow_name}: {}", flow.input);
+                self.watch_request_sender.send(request).await?
+            };
+        }
+        Ok(())
+    }
+
+    async fn publish_result(&mut self, result: FlowResult) -> Result<(), RuntimeError> {
+        match result {
+            FlowResult::Ok {
+                flow,
+                messages,
+                output,
+            } => self.publish(flow, messages, &output).await,
+            FlowResult::Err {
+                flow,
+                error,
+                output,
+            } => self.publish_error(flow, error, &output).await,
+        }
+    }
+
+    async fn publish(
+        &mut self,
+        flow: String,
+        messages: Vec<Message>,
+        output: &FlowOutput,
+    ) -> Result<(), RuntimeError> {
+        match output {
+            FlowOutput::Mqtt { topic } => {
+                for mut message in messages {
+                    if let Some(output_topic) = topic {
+                        message.topic = output_topic.name.clone();
+                    }
+                    match MqttMessage::try_from(message) {
+                        Ok(message) => self.mqtt_sender.send(message).await?,
+                        Err(err) => {
+                            error!(target: "flows", "{flow}: cannot publish transformed message: {err}")
                         }
                     }
                 }
-                Err(err) => {
-                    error!(target: "flows", "{flow_id}: {err}");
+            }
+            FlowOutput::File { path } => {
+                let Ok(mut file) = tokio::fs::File::options()
+                    .create(true)
+                    .append(true)
+                    .open(path)
+                    .await
+                    .map_err(|err| {
+                        error!(target: "flows", "{flow}: cannot open {path}: {err}");
+                    })
+                else {
+                    return Ok(());
+                };
+                for message in messages {
+                    if let Err(err) = file.write_all(format!("{message}\n").as_bytes()).await {
```

Comment:
As we're writing potentially many times to the file, I think we might want to consider using `tokio::io::BufWriter`